[bug] Custom AI preset: Connection Test sends OPTIONS instead of GET to local servers (CORS preflight)

**Describe the bug**
The Connection Test fails for a Custom AI preset pointing to a local server such as LM Studio. Instead of sending a GET /v1/models request, the app sends an OPTIONS /v1/models preflight request first, which local servers do not expect and may not handle correctly — causing the connection test to fail even when the
  server is running and reachable.

**Steps to reproduce**
  1. Install and run LM Studio with a local model loaded (default port 1234)
  2. Go to Settings → AI Presets → create a Custom preset with URL http://localhost:1234/v1
  3. Click Run Connection Test
  4. Observe the test fails

**Error log**

  LM Studio logs show:

  `2026-03-29 16:36:43 [ERROR] Unexpected endpoint or method. (OPTIONS /v1/models). Returning 200 anyway`

**Root cause**

In `components/settings/ai-presets.tsx`, the `runDiagnostics()` function uses the browser's native fetch for the models endpoint check:

`const modelsFetchFn = fetch;`

When custom headers such as `Authorization: Bearer ...` are included and the target is a different origin (e.g. localhost:1234), the browser automatically sends a CORS preflight `OPTIONS` request before the actual `GET`. Local servers like LM Studio do not implement CORS preflight handlers on `/v1/models`.

`tauriFetch (from @tauri-apps/plugin-http)` is already imported in the file but not used here. It makes native OS-level HTTP requests that bypass CORS entirely.

  // fix: replace this line
 `const modelsFetchFn = fetch;`
  // with
  `const modelsFetchFn = tauriFetch;`

**System info**

  - OS: macOS 25.3.0
  - Screenpipe version: 2.2.293

  **Additional context**

This likely affects any local OpenAI-compatible server (Ollama custom ports, Jan, AnythingLLM, etc.) when an API key is configured, since the Authorization header is what triggers the CORS preflight.

--- a/apps/screenpipe-app-tauri/components/settings/ai-presets.tsx
+++ b/apps/screenpipe-app-tauri/components/settings/ai-presets.tsx
@@ -106,6 +106,16 @@ const formatPresetName = (name: string): string => {
   return name;
 };
 
+const isLocalhostUrl = (url?: string): boolean => {
+  if (!url) return false;
+  try {
+    const hostname = new URL(url).hostname.toLowerCase();
+    return hostname === "localhost" || hostname === "127.0.0.1" || hostname === "::1";
+  } catch {
+    return false;
+  }
+};
+
 type DiagnosticStatus = "pass" | "fail" | "skip" | "pending" | "running";
 
 interface DiagnosticStepResult {
@@ -585,7 +595,11 @@ const AISection = ({
         chat: { status: "running", message: "Sending test message..." },
       }));
     } else {
-      const modelsFetchFn = fetch;
+      // Local custom providers often do not implement browser CORS preflight on /models.
+      const modelsFetchFn =
+        settingsPreset?.provider === "custom" && isLocalhostUrl(settingsPreset?.url)
+          ? tauriFetch
+          : fetch;
       try {
         modelsResponse = await modelsFetchFn(modelsUrl, {
           headers,
@@ -832,7 +846,8 @@ const AISection = ({
           break;
         case "custom":
           try {
-            const customResponse = await fetch(
+            const customFetchFn = isLocalhostUrl(settingsPreset?.url) ? tauriFetch : fetch;
+            const customResponse = await customFetchFn(
               `${settingsPreset?.url}/models`,
               {
                 headers: settingsPreset.apiKey

